src/transaction/boot_sr.c

```diff
@@ -2505,7 +2505,24 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
 #endif /* SERVER_MODE */
 
   // after recovery we can boot vacuum
-  error_code = vacuum_boot (thread_p);
+  if (get_server_type () == SERVER_TYPE_TRANSACTION)
```

Comment:
- the code can be made more compact, by treating error case immediately after the vacuum_boot () call.
- if an error occurs, the error is already logged and boot is aborted. the debug log is not necessary
- the vacuum is started message should be on `error_code == NO_ERROR` case.
```c
  if (get_server_type () == SERVER_TYPE_TRANSACTION)
    {
      error_code = vacuum_boot (thread_p);
      if (error_code != NO_ERROR)
	{
	  ASSERT_ERROR ();
	  goto error;
	}
      else
	{
	  er_log_debug (ARG_FILE_LINE, "Vacuum was started on the transaction server.");
	}
    }
  else
    {
      er_log_debug (ARG_FILE_LINE, "Vacuum was not started on the page server.");
    }
```